extension/src/popup/views/AccountHistory/index.tsx

```diff
@@ -88,75 +67,79 @@ export const AccountHistory = () => {
-        const isSorobanXfer =
-          getAttrsFromSorobanHorizonOp(operation, networkDetails)?.fnName ===
-          SorobanTokenInterface.transfer;
-        const isSwap = getIsSwap(operation);
-        const isCreateExternalAccount =
-          operation.type ===
-            Horizon.HorizonApi.OperationResponseType.createAccount &&
-          operation.account !== publicKey;
-        const isDustPayment =
-          isPayment &&
-          "asset_type" in operation &&
-          operation.asset_type === "native" &&
-          "to" in operation &&
-          operation.to === publicKey &&
-          "amount" in operation &&
-          new BigNumber(operation.amount).lte(new BigNumber(0.1));
-        const historyOperation = {
-          ...operation,
-          isPayment,
-          isSwap,
-          isCreateExternalAccount,
-        };
-
-        if (isDustPayment && isHideDustEnabled) {
-          return;
-        }
-
-        if ((isPayment || isSorobanXfer) && !isSwap) {
-          if (operation.source_account === publicKey) {
-            segments[SELECTOR_OPTIONS.SENT].push(historyOperation);
-          } else if ("to" in operation && operation.to === publicKey) {
-            segments[SELECTOR_OPTIONS.RECEIVED].push(historyOperation);
+    ) =>
+      operations.reduce(
+        (
+          sections: HistorySection[],
+          operation: Horizon.ServerApi.OperationRecord,
+        ) => {
+          const isPayment = getIsPayment(operation.type);
+          const isSwap = getIsSwap(operation);
+          const isCreateExternalAccount =
+            operation.type ===
+              Horizon.HorizonApi.OperationResponseType.createAccount &&
+            operation.account !== publicKey;
+          const isDustPayment =
+            isPayment &&
+            "asset_type" in operation &&
+            operation.asset_type === "native" &&
+            "to" in operation &&
+            operation.to === publicKey &&
+            "amount" in operation &&
+            new BigNumber(operation.amount).lte(new BigNumber(0.1));
+          const parsedOperation = {
+            ...operation,
+            isPayment,
+            isSwap,
+            isCreateExternalAccount,
+          };
+
+          if (isDustPayment && isHideDustEnabled) {
+            return sections;
           }
-        }
 
-        if (isCreateExternalAccount) {
-          segments[SELECTOR_OPTIONS.SENT].push(historyOperation);
-        }
+          const date = new Date(operation.created_at);
+          const month = date.getMonth();
+          const year = date.getFullYear();
+          const monthYear = `${month}:${year}`;
 
-        segments[SELECTOR_OPTIONS.ALL].push(historyOperation);
-      });
+          // pop() is a very performant method to get the last element of an array
+          // but it actually removes the last element, so we need to make sure
+          // to put it back later.
+          const lastSection = sections.pop();
```

Comment:
maybe a safer way to do this is to do `lastSection = sections[sections.length -1]` to get the last item without transforming the array;

Reply:
https://github.com/stellar/freighter/pull/1737/commits/7bfb35047e5f463cae2155b76350c3f87990822f